### PR TITLE
initial commit of working Ogg

### PR DESCRIPTION
**Pull request by bobparadiso**
Tested with the lua snippets below.
The ogg file bgm was able to play, looping. And triggering wav and ogg sound effects at the same time worked.
Tested fine on Windows and RPi 3.

I only saw after doing this work that there was already an old ogg branch. Please merge in this code if there was a reason the other branch couldn't be used. But an rate, adding support for other audio formats in addition to WAV is important.
Thank you!

	bgm = love.audio.newSource("bgm.ogg", "stream")
	bgm:setLooping(true)
	love.audio.play(bgm)
	
	sfx1 = love.audio.newSource("sfx1.wav", "static")
	sfx2 = love.audio.newSource("sfx2.ogg", "static")
...
		if love.input.joypad("a") then
			love.audio.play(sfx1)
		end
		if love.input.joypad("b") then
			love.audio.play(sfx2)
		end

### CODE REVIEW

File: audio.c [hunk 1]
```diff
@@ -114,26 +150,46 @@ int audio_newSource(lua_State *L)
       strlcpy(fullpath, settings.gamedir, sizeof(fullpath));
       strlcat(fullpath, path, sizeof(fullpath));
 
-      FILE *fp = fopen(fullpath, "rb");
-      if (!fp)
-         return -1;
+      //get file extension
+      char ext[PATH_MAX_LENGTH];
+      strcpy(ext, path_get_extension(path));
```
Comment (kivutar) [inline audio.c:155]: on OSX:

```
audio.c:165:19: error: implicit declaration of function 'path_get_extension' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      strcpy(ext, path_get_extension(path));
```

**Review by kivutar — CHANGES_REQUESTED** [review]
Thanks a lot for this, I'm starting a review

File: audio.c (hunk 1 above)
Comment (kivutar) [reply audio.c:155]: You need

```
+#include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
+#include <file/file_path.h>
+#include <audio/conversion/float_to_s16.h>
```

File: audio.c [hunk 2]
```diff
@@ -303,19 +359,49 @@ int source_gc(lua_State *L)
...
+   if (self->sndta.fp)
+   {
+      success = fseek(self->sndta.fp, WAV_HEADER_SIZE, SEEK_SET) == 0;
+      if (success)
+        self->state = AUDIO_STOPPED;
+   }
+   //OGG file
+   else if (self->oggData)
+   {
+      printf("audio_stop");
```
Comment (kivutar) [inline audio.c:399]: Please remove this log too

**Review by kivutar — CHANGES_REQUESTED** [review]

File: audio.c (hunk 1 above)
Comment (bobparadiso) [reply audio.c:158]: done

File: audio.c (hunk 2 above)
Comment (bobparadiso) [reply audio.c]: and done. Thanks!

**Comment by bobparadiso** [comment]
@kivutar Please let me know if there's anything else I can do to help get this feature in. Thank you very much!

**Review by kivutar — APPROVED** [review]

**Comment by kivutar** [comment]
Thank you @bobparadiso 
It's very nice to finally have ogg support

**Pull request merged**